### ISSUE TITLE
Local cache hits max filename length on Windows

### ISSUE THREAD

**Issue by VincentBailly** [microsoft/backfill#316]
The files stored in the local cache can easily hit the max filename length.
This is because the local cache reproduces the folder structure of the cached package in a location which already has a long path.

The behavior of the backfill cache when this happens in unpredictable, backfill may fail, backfill may silently fail to restore the impacted files.

More investigation is needed to observe what are the exact consequences of this bug.

**Comment by bweggersen** [comment]
Thanks for the report. This is specific for Windows, right?

**Comment by VincentBailly** [comment]
Yes, It was specified in the title but not in the description, I updated the description to make it harder to miss.
